.github/workflows/ci.yml

```diff
@@ -16,64 +16,103 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  setup-python-environment:
+  setup-python-environment-ubuntu:
     if: github.event.pull_request.draft == false
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+      fail-fast: false
     uses: ./.github/workflows/setup-python-environment.yml
+    with:
+      python-version: ${{ matrix.python-version }}
+      os: ${{ matrix.os }}
     secrets: inherit
 
-  lint-code:
-    needs: setup-python-environment
-    uses: ./.github/workflows/lint.yml
+  setup-python-environment-windows:
+    if: github.event.pull_request.draft == false
+    strategy:
+      matrix:
+        os: [windows-latest]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+      fail-fast: false
+    uses: ./.github/workflows/setup-python-environment.yml
+    with:
+      python-version: ${{ matrix.python-version }}
+      os: ${{ matrix.os }}
     secrets: inherit
 
-  unit-test:
-    needs: setup-python-environment
-    uses: ./.github/workflows/unit-test.yml
+  setup-python-environment-macos:
+    if: github.event.pull_request.draft == false
+    strategy:
+      matrix:
+        os: [macos-latest]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+      fail-fast: false
+    uses: ./.github/workflows/setup-python-environment.yml
+    with:
+      python-version: ${{ matrix.python-version }}
+      os: ${{ matrix.os }}
     secrets: inherit
 
-  integration-test:
-    needs: setup-python-environment
-    # Exit if its a commit from Gitbook
-    if: ${{ ! startsWith(github.event.head_commit.message, 'GitBook:') }}
+  lint-unit-test-ubuntu:
+    needs: setup-python-environment-ubuntu
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
```

Comment:
is it possible to declare these versions in one place somewhere? Just thinking about when we have to update this + needing to change in multiple places.

Reply:
I tried a bunch of things, including YAML anchors, but hit this 3+ years old open request to GitHub to support them that is still in limbo and then I just gave up: https://github.com/actions/runner/issues/1182